TrackingCodeCustomizer.php

```diff
@@ -74,29 +77,119 @@ public function getListHooksRegistered()
     public function applyTrackingCodeCustomizations(&$sysparams,$parameters){
         
         $originalSysparams = $sysparams;
+        $storedSettings = $this->getSettings();
         
-        $settings = API::getInstance();
-        
-        $storedSettings = $settings->getSettings();
-        
-        if(array_key_exists("options", $storedSettings))
-                $storedSettings["options"] .= $sysparams["options"];
+        if(array_key_exists("options", $storedSettings)) {
+            $storedSettings["options"] .= $sysparams["options"];
+        }
 
-        if(array_key_exists("optionsBeforeTrackerUrl", $storedSettings))
-                $storedSettings["optionsBeforeTrackerUrl"] .=$sysparams["optionsBeforeTrackerUrl"];
-        
-        $sysparams = array_merge($sysparams,$storedSettings);
+        if(array_key_exists("optionsBeforeTrackerUrl", $storedSettings)) {
+            $storedSettings["optionsBeforeTrackerUrl"] .= $sysparams["optionsBeforeTrackerUrl"];
+        }
+
+        $sysparams = array_merge($sysparams, $storedSettings);
         
         foreach($sysparams as $key => $value){
-            
             $sysparams[$key] =  $this->replaceTokens($value,$originalSysparams,$sysparams);
         }
     
     }
-    
+
+    /**
+     * @param $subject
+     * @param $originalSysparams
+     * @param $sysparams
+     * @return mixed
+     */
     private function replaceTokens($subject,$originalSysparams,$sysparams){
         $output = str_replace(array_map(function($item){return '{$original_'.$item.'}';},array_keys($originalSysparams)),array_values($originalSysparams),$subject);
         $output = str_replace(array_map(function($item){return '{$'.$item.'}';},array_keys($sysparams)),array_values($sysparams),$output);
         return $output;
     }
+
+    /**
+     * @return array
+     */
+    protected function getSettings() {
+        $settings = API::getInstance();
+        $storedSettings = $settings->getSettings();
+        return $storedSettings;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getTrackingCodeFile() {
+        return $trackingCodeFile = PIWIK_INCLUDE_PATH . "/piwik.js";
+    }
+
+    /**
+     *
+     */
+    public function afterUpdate() {
+
+        // cleanup old original file
+        $oldOriginalCodeFile = $this->getTrackingCodeFile() . ".original";
+        if(file_exists($oldOriginalCodeFile)) {
+            unlink($oldOriginalCodeFile);
+        }
+
+        $this->rewriteJavascriptFiles();
+    }
+
+    /**
+     *
+     */
+    public function rewriteJavascriptFiles() {
+        $trackingCodeFile = $this->getTrackingCodeFile();
+        $trackingCodeFileOriginal = $trackingCodeFile . ".original";
+
+        if(!file_exists($trackingCodeFileOriginal)) {
+            copy($trackingCodeFile, $trackingCodeFileOriginal);
```

Comment:
This assumes webserver has write permissions to file. Need to check permissions or handle failure and provide instruction to user. User would need to either SSH in to do the process manually or set write perm for web server process.